grakn-engine/src/main/java/ai/grakn/engine/controller/api/AttributeTypeController.java

```diff
@@ -0,0 +1,144 @@
+            String jsonAttributeTypeLabel = attributeType.getLabel().getValue();
+            String jsonAttributeTypeDataType = toString(attributeType.getDataType());
+            LOG.info("postAttributeType - attribute type " + jsonAttributeTypeLabel +
+                " of type " + jsonAttributeTypeDataType + " with id " + jsonConceptId + " added successfully. request processed.");
+            response.status(HttpStatus.SC_OK);
+            Json responseBody = attributeTypeJson(jsonConceptId, jsonAttributeTypeLabel, jsonAttributeTypeDataType);
+            return responseBody;
+        }
+    }
+
+    private Json getAttributeType(Request request, Response response) {
+        LOG.info("getAttributeType - request received.");
+        String attributeTypeLabel = mandatoryPathParameter(request, "attributeTypeLabel");
+        String keyspace = mandatoryQueryParameter(request, KEYSPACE);
+        LOG.info("getAttributeType - attempting to find attributeType " + attributeTypeLabel + " in keyspace " + keyspace);
+        try (GraknTx tx = factory.tx(Keyspace.of(keyspace), GraknTxType.READ)) {
+            Optional<AttributeType> attributeType = Optional.ofNullable(tx.getAttributeType(attributeTypeLabel));
+            if (attributeType.isPresent()) {
+                String jsonConceptId = attributeType.get().getId().getValue();
+                String jsonAttributeTypeLabel = attributeType.get().getLabel().getValue();
+                String jsonAttributeTypeDataType = toString(attributeType.get().getDataType());
+                response.status(HttpStatus.SC_OK);
+                Json responseBody = attributeTypeJson(jsonConceptId, jsonAttributeTypeLabel, jsonAttributeTypeDataType);
+                LOG.info("getAttributeType - attributeType found - " + jsonConceptId + ", " +
+                    jsonAttributeTypeLabel + ", " + jsonAttributeTypeDataType + ". request processed.");
+                return responseBody;
+            } else {
+                response.status(HttpStatus.SC_BAD_REQUEST);
+                LOG.info("getAttributeType - attributeType NOT found. request processed.");
+                return Json.nil();
+            }
+        }
+    }
+
+    private AttributeType.DataType<?> fromString(String dataType) {
+        if (dataType.equals("string")) {
+            return AttributeType.DataType.STRING;
+        } else if (dataType.equals("double")) {
+            return AttributeType.DataType.DOUBLE;
+        } else if (dataType.equals("long")) {
+            return AttributeType.DataType.LONG;
+        } else if (dataType.equals("boolean")) {
+            return AttributeType.DataType.BOOLEAN;
+        } else {
+            throw GraknServerException.invalidQueryExplaination("invalid datatype supplied: '" + dataType + "'");
+        }
+    }
+
+    private String toString(AttributeType.DataType<?> dataType) {
+        if (dataType.equals(AttributeType.DataType.STRING)) {
```

Comment:
As far as I remember it would return `java.lang.String` instead of just `string`, but let me double check.

Reply:
Yup, confirmed that we will need `DataType fromString()` / `toString(DataType)` methods. If we rely on `dataType.getName` it will return `java.lang.String`, `java.lang.Boolean` etc which basically exposes implementation details and therefore inappropriate for a public API.

Reply:
ditto for here as well

Reply:
@lolski good news! I already have this for Graql in `QueryParser#DATA_TYPES`. We should probably make it a method in `AttributeType.DataType<?>` though.

Reply:
@aelred cool! i can use that then. is there ever the need of using the value returned by `dataType.getName`? otherwise, later on, we should just have one method for consistency's sake

Reply:
Guys there might be an easier approach. Maybe we can just change the keys of the `SUPPORTED_TYPES` to be the simple name of the class. So instead of of `java.Lang.String` it will just be `String`. I don't think there are any negatives from that action. 

Might even let you get rid of your helper in Graql @aelred

Reply:
IMHO it's still better having the datatype specified somewhere as what @aelred do. Otherwise, for `date`, it will be `LocalDateTime` instead.

Reply:
@fppt I think that's a good idea. We can't call it `String` though. It would have to be the Graql keyword, which is `long`/`double`/`date`/`boolean`/`string`.

Reply:
@aelred Convert it to lowercase in the map then. Just trying to reduce the duplicate code.

Reply:
I am saying we maintain the public facing types `long/double/date/boolean/string` i.e we don't use `LocalDateTime `

Reply:
fixed in e80ab37071edd14a98ec87ce56c07873ffc35e56